segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -518,14 +546,22 @@ public CompletableFuture<Void> seal(SegmentHandle handle, Duration timeout) {
     @Override
     public CompletableFuture<Void> concat(SegmentHandle targetHandle, long offset, String sourceSegment, Duration timeout) {
         checkInitialized();
+
+        Exception ex = null;
         if (null == targetHandle) {
-            return CompletableFuture.failedFuture(new IllegalArgumentException("handle must not be null"));
-        }
-        if (null == sourceSegment) {
-            return CompletableFuture.failedFuture(new IllegalArgumentException("sourceSegment must not be null"));
+             ex = new IllegalArgumentException("handle must not be null");
+        } else if (null == sourceSegment) {
```

Comment:
is there a scope for refactoring these common set of preconditions?

Reply:
I think I had tried that earlier and the refactored code looks pretty much the same (because it needs to return). But definitely will see if we can do this better.

Reply:
done